cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1011,7 +983,7 @@ if(ARROW_BUILD_INTEGRATION
   set(ARROW_USE_BOOST TRUE)
   set(ARROW_BOOST_REQUIRE_LIBRARY TRUE)
 elseif(ARROW_GANDIVA
-       OR (ARROW_WITH_THRIFT AND THRIFT_REQUIRES_BOOST)
+       OR ARROW_WITH_THRIFT
```

Comment:
So boost is always required by Thrift nevertheless?

Reply:
Yes. The condition was wrong. We needs Boost to use Thrift.

We need Thrift only when we use Parquet or `arrow/dbi/hiveserver2/`.
Parquet uses `thrift/transport/TBufferTransports.h`: https://github.com/apache/arrow/blob/master/cpp/src/parquet/thrift_internal.h#L43
`thrift/transport/TBufferTransports.h` uses Boost: https://github.com/apache/thrift/blob/master/lib/cpp/src/thrift/transport/TBufferTransports.h#L26